File: src/main/java/ejercicioJava/SistemaGestionBiblioteca/Biblioteca.java
package ejercicioJava.SistemaGestionBiblioteca;

import java.util.ArrayList;
import java.util.List;

public class Biblioteca {

    public List<Libro> libros = new ArrayList<>();

    public void agregarLibros(Libro libro) {
        if (!libros.contains(libro)) {
            libros.add(libro);
        }
    }

    private void registrarUsuarioLibro(Usuario usuario, Libro libro) {
        usuario.setLibro(libro);
        libro.setPrestado(true);
    }

    public boolean prestarUsuario(Usuario usuario, Libro libro) {
        if (!libro.isPrestado()) {
            registrarUsuarioLibro(usuario, libro);
            return true;
        }
        return false;
    }

    public boolean devolverLibro(Usuario usuario) {
        if (usuario.getLibro() != null) {
            devolverUsuarioLibro(usuario, usuario.getLibro());
            return true;
        }
        return false;
    }

    private void devolverUsuarioLibro(Usuario usuario, Libro libro) {
        usuario.setLibro(null);
        libro.setPrestado(false);
    }
}

File: src/main/kotlin/ejercicioKotlin/sistemaGestionBiblioteca/Biblioteca.kt
package org.example.ejercicioKotlin.sistemaGestionBiblioteca

class Biblioteca() {

    private val libros:MutableList<Libro> = mutableListOf()

    fun agregarLibros(libro: Libro):Boolean {
        if (libro !in libros) {
            libros.add(libro)
            return true
        }
        return false
    }

    private fun registrarUsuarioLibro(usuario: Usuario, libro: Libro){
        usuario.libro = libro
        libro.prestado = true
    }

    fun prestrarUsuario(usuario: Usuario, libro: Libro):Boolean {
        if (!libro.prestado) {
            registrarUsuarioLibro(usuario, libro)
            return true
        }
        return false
    }

    fun devolverLibro(usuario: Usuario) :Boolean {
        if (usuario.libro != null) {
            devolverUsuarioLibro(usuario, usuario.libro ?: Libro(""))
            return true
        }
        return false
    }


    private fun devolverUsuarioLibro(usuario: Usuario, libro: Libro) {
        usuario.libro = null
        libro.prestado = false
    }

}